pkg/maps/lbmap/maglev_inner_map.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lbmap
+
+import (
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/ebpf"
+)
+
+// maglevInnerMap is the internal representation of a maglev inner map.
+type maglevInnerMap struct {
+	*ebpf.Map
+	TableSize uint32
+}
+
+// MaglevInnerKey is the key of a maglev inner map.
+type MaglevInnerKey struct {
+	Zero uint32
+}
+
+// MaglevInnerVal is the value of a maglev inner map.
+type MaglevInnerVal struct {
+	BackendIDs []uint16
+}
+
+// NewMaglevInnerMap returns a new object representing a maglev inner map.
+func NewMaglevInnerMap(name string, tableSize uint32) (*maglevInnerMap, *ebpf.MapSpec, error) {
+	spec := &ebpf.MapSpec{
+		Name:       name,
+		Type:       ebpf.Array,
+		KeySize:    uint32(unsafe.Sizeof(MaglevInnerKey{})),
+		ValueSize:  uint32(unsafe.Sizeof(uint16(0)) * uintptr(tableSize)),
```

Comment:
nit: `unsafe.Sizeof(uint16(0))` will always have value 2, no? I think it's safe to use a constant here.

Reply:
correct, but I'd like to keep it explicit to save a comment/avoid in the future wondering why `ValueSize` is `2 * uintptr(tableSize)`